src/sqlfluff/core/linter/linter.py

```diff
@@ -506,6 +507,10 @@ def lint_fix_parsed(
             )
 
             for crawler in progress_bar_crawler:
+                # Performance: After first loop pass, skip rules that don't do fixes.
+                if fix and loop > 0 and not is_fix_compatible(crawler):
+                    continue
```

Comment:
About 1/4 of the rules aren't fix compatible. But we were executing them on every linter pass. Wasteful!